test/IDE/complete_constructor.swift

```diff
@@ -119,7 +119,6 @@ func testExplicitConstructors3P() {
 // EXPLICIT_CONSTRUCTORS_3P: Begin completions
 // EXPLICIT_CONSTRUCTORS_3P-DAG: Decl[Constructor]/CurrNominal: ['(']{#(a): Int#}[')'][#ExplicitConstructors3#]{{; name=.+$}}
 // EXPLICIT_CONSTRUCTORS_3P-DAG: Decl[Constructor]/CurrNominal: ['(']{#a: Int#}, {#b: Float#}[')'][#ExplicitConstructors3#]{{; name=.+$}}
-// EXPLICIT_CONSTRUCTORS_3P-DAG: Decl[FreeFunction]/CurrModule/NotRecommended/TypeRelation[Invalid]: freeFunc()[#Void#]{{; name=.+$}}
```

Comment:
So I am wondering how the code completion changes (which I haven't looked at closely) are related to the type cache. I would expect no behavioral changes from making the type checker store types in a different place.